src/librustc_typeck/diagnostics.rs

```diff
@@ -1130,6 +1244,96 @@ impl Bytes { ... } // error, same as above
 ```
 "##,
 
+E0117: r##"
+You got this error because because you tried to implement a foreign
+trait on a foreign (with maybe a foreign type parameter). Erroneous
```

Comment:
foreign trait **for** a foreign type.

Foreign trait is enough, because traits parametrised by local types are not foreign traits anymore.